Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithm.cpp

```diff
@@ -223,44 +235,45 @@ PsatdAlgorithm::pushSpectralFields (SpectralFieldData& f) const
             const Complex X4 = (is_galilean) ? X4_arr(i,j,k) : - S_ck / PhysConst::ep0;
             const Complex T2 = (is_galilean) ? T2_arr(i,j,k) : 1.0_rt;
 
-            // Update equations for E in the formulation with rho
-            // T2 = 1 always with standard PSATD (zero Galilean velocity)
-
+            // Shortcuts for the values of rho
+            Complex rho_old, rho_new;
             if (update_with_rho)
             {
-                fields(i,j,k,Idx.Ex) = T2 * C * Ex_old
-                                       + I * c2 * T2 * S_ck * (ky * Bz_old - kz * By_old)
-                                       + X4 * Jx - I * (X2 * rho_new - T2 * X3 * rho_old) * kx;
+                rho_old = fields(i,j,k,Idx.rho_old);
+                rho_new = fields(i,j,k,Idx.rho_new);
+            }
+            else // update_with_rho = 0
+            {
+                const amrex::Real kc_dot_vg = kx_c*vgx + ky_c*vgy + kz_c*vgz;
+                const Complex k_dot_E = kx*Ex_old + ky*Ey_old + kz*Ez_old;
+                const Complex k_dot_J = kx*Jx + ky*Jy + kz*Jz;
 
-                fields(i,j,k,Idx.Ey) = T2 * C * Ey_old
-                                       + I * c2 * T2 * S_ck * (kz * Bx_old - kx * Bz_old)
-                                       + X4 * Jy - I * (X2 * rho_new - T2 * X3 * rho_old) * ky;
+                rho_old = I*ep0*k_dot_E;
```

Comment:
This is the expression of `rho_old` in terms of div(E) in spectral space, for both standard PSATD and Galilean PSATD.